chart/templates/webserver/webserver-ingress.yaml

```diff
@@ -39,12 +39,27 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
-  {{- if .Values.ingress.web.tls.enabled }}
+  {{- if and .Values.ingress.web.hosts (.Values.ingress.web.hosts | first | kindIs "string" | not) }}
+  {{- $anyTlsHosts := false -}}
+  {{- range .Values.ingress.web.hosts }}
+  {{- if .tls.enabled }}
```

Comment:
i think there is something wrong here.  i tried rendering this config (the 1.20 list[str] behavior)

```
ingress:
  enabled: true
  web:
    # host: "hi.com"
    hosts:
    - name: "yo.com"
    #   # configs for web Ingress TLS
    #   tls:
    #     # Enable TLS termination for the web Ingress
    #     enabled: false
    #     # the name of a pre-created Secret containing a TLS private key and certificate
    #     secretName: ""
```

and i get this error

```
❯ helm template ~/code/airflow/chart -f ingress-config.yaml -s templates/webserver/webserver-ingress.yaml
Error: template: airflow/templates/webserver/webserver-ingress.yaml:45:13: executing "airflow/templates/webserver/webserver-ingress.yaml" at <.tls.enabled>: nil pointer evaluating interface {}.enable
```

and if i'm actually catching something then there are probably similar issues where this logic is repeated

Reply:
The change that is being made as part of this PR is moving hosts from an array of strings ie list of hostnames, to an object containing host data: hostname and tls config.  I have assumed those are required here, but perhaps I should change that.

Reply:
I assumed this change was supposed to be backwards compatible, so that list[str] should be deprecated but still work. 

It's possible to break things in a major release though, and @jedcunningham mentioned possibly doing that sooner than later and cleaning up some backcompat.  @jedcunningham should we keep backward compatibility in this one or just target next major?

Reply:
This update is very easy to make, so I just went ahead and did it.  On that note the lack of short circuit boolean evaluation in go templates is aggravating.

Reply:
Great thank you I'll give it another look tomorrow or Monday